pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -85,6 +87,20 @@ func (proxy *DataSourceProxy) HandleRequest() {
 
 	proxy.logRequest()
 
+	span, ctx := opentracing.StartSpanFromContext(proxy.ctx.Req.Context(), "datasource reverse proxy")
+	proxy.ctx.Req.Request = proxy.ctx.Req.WithContext(ctx)
+
+	defer span.Finish()
+	span.SetTag("datasource_id", proxy.ds.Id)
+	span.SetTag("datasource_type", proxy.ds.Type)
+	span.SetTag("user_id", proxy.ctx.SignedInUser.UserId)
+	span.SetTag("org_id", proxy.ctx.SignedInUser.OrgId)
+
+	opentracing.GlobalTracer().Inject(
+		span.Context(),
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(proxy.ctx.Req.Request.Header))
+
 	reverseProxy.ServeHTTP(proxy.ctx.Resp, proxy.ctx.Req.Request)
```

Comment:
if this call times out, or there's an error or something, will it be marked in the span?

Reply:
yes. Grafana will return the same request as the upstream and tag it in https://github.com/grafana/grafana/blob/dist_tracing/pkg/middleware/request_tracing.go#L29

Reply:
i meant a bool `error` tag with value true (which is standard OT) and gets rendered specially in jaeger (in red)

Reply:
@Dieterbe Hm. 

Should Grafana mark all 4xx and 5xx responses as errors or should downstream systems set the error? As close to the source as possible. @yurishkuro make you can give us your thoughts.

Reply:
downstream systems may not implement OT though. if they do, the whole trace will be marked erroreous, if they don't, grafana should do it.

Reply:
you should definitely set `error=true` on your side if you consider the span as error. Sometimes there's contention with whether 4xx should be considered errors, especially for the server-side spans. I personally think they should be, and they can always be separated from 5xx because the status code is logged separately.

Reply:
All requests that return >=400 are now marked as error. I think it makes sense for downstream system and Grafana itself. 404 is the only one that made me hesitate a little bit but if someone is expecting a specific dashboards and cannot find it I think its a failed request.